mitosheet/src/components/taskpanes/SplitTextToColumns/SplitTextToColumnsTaskpane.tsx

```diff
@@ -0,0 +1,302 @@
+import React, { useEffect, useState } from "react";
+import MitoAPI from "../../../jupyter/api";
+import { AnalysisData, ColumnID, GridState, SheetData, SplitTextToColumnsParams, StepType, UIState } from "../../../types"
+import DefaultEmptyTaskpane from "../DefaultTaskpane/DefaultEmptyTaskpane";
+import DefaultTaskpane from "../DefaultTaskpane/DefaultTaskpane";
+import DefaultTaskpaneBody from "../DefaultTaskpane/DefaultTaskpaneBody";
+import DefaultTaskpaneHeader from "../DefaultTaskpane/DefaultTaskpaneHeader";
+import Row from "../../spacing/Row";
+import Col from "../../spacing/Col";
+import Select from "../../elements/Select";
+import DropdownItem from "../../elements/DropdownItem";
+import { getDisplayColumnHeader } from "../../../utils/columnHeaders";
+import MultiSelectButtons from "../../elements/MultiSelectButtons";
+import useSendEditOnClick from "../../../hooks/useSendEditOnClick";
+import TextButton from "../../elements/TextButton";
+import Input from "../../elements/Input";
+import { useDebouncedEffect } from "../../../hooks/useDebouncedEffect";
+import '../../../../css/taskpanes/SplitTextToColumns/SplitTextToColumns.css'
+import DefaultTaskpaneFooter from "../DefaultTaskpane/DefaultTaskpaneFooter";
+
+interface SplitTextToColumnsTaskpaneProps {
+    mitoAPI: MitoAPI;
+    analysisData: AnalysisData;
+    sheetDataArray: SheetData[];
+    gridState: GridState;
+    setUIState: React.Dispatch<React.SetStateAction<UIState>>;
+    dfNames: string[];
+    startingColumnID?: ColumnID
```

Comment:
Just based on how much work you're doing at the start of this file to find the correct column id, why not just make this not optional, ever?

No matter where you're setting the taskpane from, you know the column ID that is selected (even in the actions, this work is already done). 

It would chop out so much code at the start of this file that is just duplicated work anyways.